### PR TITLE
Removal of MAX_PATH limit on Windows

### PR DESCRIPTION
**Pull request by CookiePLMonster**
This PR lifts `MAX_PATH` limit from all code places which still used it. Most of current instances of it could be trivially removed either by calling functions twice (first to determine required buffer size, then to do actual work) or by replacing them with their memory-allocating counterparts.

The end result is a complete removal of `kMaxPath` constant on Windows. It's now unused and I think it's a good idea not to keep it to encourage not using any hardcoded constants for functions operating on paths.

### CODE REVIEW

File: src/xenia/gpu/shader.cc [hunk 1]
```diff
@@ -49,41 +49,58 @@ std::pair<std::string, std::string> Shader::Dump(const std::string& base_path,
...
-      fprintf(f, "// ");
-      fwrite(ucode_disasm_ptr, 1, line_end - ucode_disasm_ptr + 1, f);
-      ucode_disasm_ptr = line_end + 1;
-    }
-    fprintf(f, "\n\n");
-    if (!host_disassembly_.empty()) {
-      fprintf(f, "\n\n/*\n%s\n*/\n", host_disassembly_.c_str());
+  std::string txt_file_name;
+  int required_size =
+      std::snprintf(nullptr, 0, "%s/shader_%s_%.16" PRIX64 ".%s",
```
Comment (gibbed) [inline src/xenia/gpu/shader.cc]: The way you ended up changing it here is excessively verbose. I think it would be better to `std::snprintf` the data hash into a static buffer then concatenate the string as normal instead of this giant repetition of code.

File: src/xenia/base/platform.h [hunk 2]
```diff
@@ -95,7 +95,6 @@ namespace xe {
 #if XE_PLATFORM_WIN32
 const char kPathSeparator = '\\';
 const wchar_t kWPathSeparator = L'\\';
-const size_t kMaxPath = 260;  // _MAX_PATH
```
Comment (gibbed) [inline src/xenia/base/platform.h]: Please split this into a separate commit.

File: src/xenia/gpu/shader.cc (hunk 1 above)
Comment (CookiePLMonster) [reply src/xenia/gpu/shader.cc]: Can a reliable assumption be made about max length of this hash? If so, I can of course do it.

Comment (gibbed) [reply src/xenia/gpu/shader.cc]: `"%.16" PRIX64` is `%.16llx` -- so yes.

Comment (CookiePLMonster) [reply src/xenia/gpu/shader.cc]: Yes, but `path_prefix` doesn't seem to have size defined, so I would only save on reformatting `ucode_data_hash_` and maybe extension. Is it worth it?

Comment (CookiePLMonster) [reply src/xenia/gpu/shader.cc]: Nevermind, gave this a second look after a while and now I get what do you mean!

**Comment by CookiePLMonster** [comment]
Concerns addressed, commits reworded.

File: src/xenia/gpu/shader.cc [hunk 3]
```diff
@@ -49,12 +60,20 @@ std::pair<std::string, std::string> Shader::Dump(const std::string& base_path,
...
   }
 
-  char txt_file_name[kMaxPath];
-  std::snprintf(txt_file_name, xe::countof(txt_file_name),
-                "%s/shader_%s_%.16" PRIX64 ".%s", base_path.c_str(),
-                path_prefix, ucode_data_hash_,
-                shader_type_ == ShaderType::kVertex ? "vert" : "frag");
-  FILE* f = fopen(txt_file_name, "wb");
+  const std::string file_name_no_extension =
+      MakeShaderDumpPath(base_path, path_prefix, ucode_data_hash_);
```
Comment (DrChat) [inline src/xenia/gpu/shader.cc]: Use `xe::format_string` for this instead

**Comment by CookiePLMonster** [comment]
Rebased and addressed review comments.
